### PR TITLE
WIP: [Notebook Scale] LTS improvements

### PR DESCRIPTION
**Pull request by kdvalin**
This PR introduces multiple lts payload  improvements for the notebook scale test.

### CODE REVIEW

File: subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/horreum_lts_store.py [hunk 1]
```diff
@@ -161,27 +161,27 @@ def build_lts_payloads() -> dict:
...
                 'ocp_version': results.sutest_ocp_version,
                 'metrics': _gather_prom_metrics(entry),
                 'thresholds': results.thresholds,
-                'config': results.test_config.yaml_file
+                'config': results.test_config.yaml_file,
+                "cluster_info": _parse_entry(entry.results.rhods_cluster_info),
             },
             "metadata": {
-                "test": "rhods-notebooks",
+                "test": results.test_config.yaml_file['tests']['identifier'],
```
Comment (kpouget) [inline subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/horreum_lts_store.py:176]:
```suggestion
                "test": results.test_config.get("tests.identifier"),
```
what happens on Horreum side if this field is empty?

File: subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/models/__init__.py [hunk 2]
```diff
@@ -8,17 +8,16 @@
 
 class NotebookScaleMetadata(matbench_models.Metadata):
     test: enums.TestName
```
Comment (kpouget) [inline subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/models/__init__.py:10]: what do you think about making this an `Union[enums.TestName, str]` to allow other values?

**Comment by kdvalin** [comment]
Tried to test this PR, bailed out with a mkdir error
https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/RHODS/job/ci-artifacts/172/execution/node/36/log/

**Comment by kpouget** [comment]
no, the actual error is above:
```

[m[31m18:43:48.327 rhods-setup:322@f23-h33-000-6018r: podman ps | grep ci-artifacts
ERRO[0000] Refreshing volume ci-artifacts: acquiring lock 0 for volume ci-artifacts: file exists 
[m[37m18:43:48.328 rhods-setup:322@f23-h33-000-6018r: regex: ci-artifacts
[m[31m18:43:48.812 rhods-setup:322@f23-h33-000-6018r: podman stop ci-artifacts
Error: no container with name or ID "ci-artifacts" found: no such container
[m[31m18:43:48.819 rhods-setup:322@f23-h33-000-6018r: aborting run due to exit code 125
```

**Comment by kpouget** [comment]
I relaunched it

**Comment by kpouget** [comment]
@kdvalin,  you get the PR tested here as well, that might be simpler that running the (many-user) scale test on the Icelake cluster

**Comment by kpouget** [comment]
```
$ cat 002_test_ci/003__plots/_matbench_generate_lts.log
INFO | Loading matrix_benchmarking.workloads.rhods-notebooks.store module ...
INFO | Loading matrix_benchmarking.workloads.rhods-notebooks.store module ... done.
INFO | Loading results ...
INFO | simple store: ignoring exit code.
INFO | No threshold found in notebook_scale_test_ods_ci.yaml|notebook_scale_test.
INFO | Loading results: done, found 1 results
INFO | Gathering sutest__container_cpu__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard
INFO | Gathering sutest__container_sum_cpu__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard
INFO | Gathering sutest__container_cpu_requests__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard
INFO | Gathering sutest__container_cpu_limits__namespace=redhat-ods-applications_pod=rhods-dashboard.*_container=rhods-dashboard
INFO | Gathering Sutest API Server Requests (server errors)
INFO | Gathering Sutest Control Plane Node CPU idle
Traceback (most recent call last):
  File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/ci-artifacts/src/subprojects/matrix-benchmarking/matrix_benchmarking/main.py", line 55, in <module>
    sys.exit(main())
  File "/opt/ci-artifacts/src/subprojects/matrix-benchmarking/matrix_benchmarking/main.py", line 47, in main
    return runnable.run()
  File "/opt/ci-artifacts/src/subprojects/matrix-benchmarking/matrix_benchmarking/parse.py", line 107, in run
    for (payload, _, __) in workload_store.build_lts_payloads():
  File "/opt/ci-artifacts/src/subprojects/matrix-benchmarking/matrix_benchmarking/workloads/rhods-notebooks/horreum_lts_store.py", line 182, in build_lts_payloads
    output: models.NotebookScalePayload = models.NotebookScalePayload.parse_obj(payload)
  File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic/main.py", line 1076, in pydantic.main.validate_model
  File "pydantic/fields.py", line 860, in pydantic.fields.ModelField.validate
pydantic.errors.ConfigError: field "payload_schema" not yet prepared so type is still a ForwardRef, you might need to call NotebookScalePayload.update_forward_refs().
---

**Comment by kpouget** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/785/pull-ci-openshift-psap-ci-artifacts-main-notebooks-light/1669500067024736256/artifacts/light/test/artifacts/003__plots/)
*  [lts_payload.json](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/785/pull-ci-openshift-psap-ci-artifacts-main-notebooks-light/1669500067024736256/artifacts/light/test/artifacts/003__plots/lts_payload.json)

File: visualizations/rhods-notebooks/horreum_lts_store.py [hunk 3]
```diff
@@ -161,27 +161,27 @@ def build_lts_payloads() -> dict:
...
             "$schema": "urn:rhods-matbench-upload:3.0.0",
             "data": {
                 "users": _decode_users(results),
                 'rhods_version': results.rhods_info.version,
                 'ocp_version': results.sutest_ocp_version,
                 'metrics': _gather_prom_metrics(entry),
                 'thresholds': results.thresholds,
-                'config': results.test_config.yaml_file
+                'config': results.test_config.yaml_file,
+                "cluster_info": _parse_entry(entry.results.rhods_cluster_info),
```
Comment (kpouget) [inline visualizations/rhods-notebooks/horreum_lts_store.py:173]: I guess `_parse_entry` could have a more meaningful name, explaining that it transforms into dictionaries 

but is this field `cluster_info` actually used in Horreum side? looking at the content, I'm not sure it's relevant to include it in the LTS payload

Comment (kpouget) [reply visualizations/rhods-notebooks/horreum_lts_store.py:173]: hum, makes sense. I guess it's the control plane nodes that matter most, but still 🤔 
I guess we can keep it then, we'll have to know this list as well in horreum to track the idleness of the control plane nodes

**Comment by kdvalin** [comment]
Messed up by accidentally rebasing.  Closing this out and reopening on a clean branch

**Pull request closed without merging**